titus-client/src/main/java/com/netflix/titus/runtime/connector/jobmanager/RemoteJobManagementClientWithKeepAlive.java

```diff
@@ -94,31 +104,44 @@ public void onCompleted() {
             );
 
             // Now emit keep alive requests periodically
-            Disposable keepAliveSubscription = Flux.interval(Duration.ofMillis(configuration.getKeepAliveIntervalMs())).subscribe(
-                    next -> {
-                        try {
-                            clientStreamObserver.onNext(ObserveJobsWithKeepAliveRequest.newBuilder()
-                                    .setKeepAliveRequest(KeepAliveRequest.newBuilder()
-                                            .setRequestId(keepAliveIdGen.getAndIncrement())
-                                            .setTimestamp(titusRuntime.getClock().wallTime())
+            Disposable keepAliveSubscription = Flux.interval(Duration.ofMillis(configuration.getKeepAliveIntervalMs()))
+                    // doOnCancel is confusing: it's called when the subscription is disposed
+                    // It should be named doOnDispose. See: https://github.com/reactor/reactor-core/issues/1240
+                    .doOnCancel(
+                            () -> ExceptionExt.doCatch(keepAliveCompleted)
+                                    .ifPresent(t -> logger.warn("Error running the keepAliveCompleted callback", t))
```

Comment:
Will this print out what the exception was?

Reply:
yes, it should, because of the second parameter